internal/extproc/translator/openai_awsbedrock.go

```diff
@@ -152,113 +160,196 @@ func (o *openAIToAWSBedrockTranslatorV1ChatCompletion) openAIToolsToBedrockToolC
+		for _, contentPart := range contents {
+			if contentPart.TextContent != nil {
+				textContentPart := contentPart.TextContent
+				chatMessage.Content = append(chatMessage.Content, &awsbedrock.ContentBlock{
+					Text: &textContentPart.Text,
+				})
+			} else if contentPart.ImageContent != nil {
+				imageContentPart := contentPart.ImageContent
+				contentType, b, err := parseDataURI(imageContentPart.ImageURL.URL)
+				if err != nil {
+					return nil, fmt.Errorf("failed to parse image URL: %s %w", imageContentPart.ImageURL.URL, err)
+				}
+				var format string
+				switch contentType {
+				case "image/png":
+					format = "png"
+				case "image/jpeg":
+					format = "jpeg"
+				case "image/gif":
+					format = "gif"
+				case "image/webp":
+					format = "webp"
+				default:
+					return nil, fmt.Errorf("unsupported image type: %s please use one of [png, jpeg, gif, webp]",
+						contentType)
+				}
+
+				chatMessage.Content = append(chatMessage.Content, &awsbedrock.ContentBlock{
+					Image: &awsbedrock.ImageBlock{
+						Format: format,
+						Source: awsbedrock.ImageSource{
+							Bytes: b, // Decoded data as bytes.
+						},
+					},
+				})
+			}
+		}
+		return chatMessage, nil
+	} else {
+		return nil, fmt.Errorf("unexpected content type")
+	}
+}
+
+// openAIMessageToBedrockMessageRoleAssistant converts openai assistant role message
+// returns a list of bedrock message and error
+// the tool content is appended to the list of bedrock message list if tool_call is in openai message
+func (o *openAIToAWSBedrockTranslatorV1ChatCompletion) openAIMessageToBedrockMessageRoleAssistant(
+	openAiMessage *openai.ChatCompletionAssistantMessageParam, role string,
+) (*awsbedrock.Message, error) {
+	var bedrockMessage *awsbedrock.Message
+	contentBlocks := make([]*awsbedrock.ContentBlock, 1)
+	if openAiMessage.Content.Type == openai.ChatCompletionAssistantMessageParamContentTypeRefusal {
+		contentBlocks[0] = &awsbedrock.ContentBlock{Text: openAiMessage.Content.Refusal}
+	} else {
+		contentBlocks[0] = &awsbedrock.ContentBlock{Text: openAiMessage.Content.Text}
+	}
+	bedrockMessage = &awsbedrock.Message{
+		Role:    role,
+		Content: contentBlocks,
+	}
+	// Process tool_calls.
+	for _, toolCall := range openAiMessage.ToolCalls {
+		bedrockMessage.Content = append(bedrockMessage.Content,
+			&awsbedrock.ContentBlock{
+				ToolUse: &awsbedrock.ToolUseBlock{
+					Name:      toolCall.Function.Name,
+					ToolUseID: toolCall.ID,
+					Input:     toolCall.Function.Arguments,
+				},
+			})
+	}
+	return bedrockMessage, nil
+}
+
+// openAIMessageToBedrockMessageRoleSystem converts openai system role message
+// returns a list of bedrock system content and error
+func (o *openAIToAWSBedrockTranslatorV1ChatCompletion) openAIMessageToBedrockMessageRoleSystem(
+	openAiMessage *openai.ChatCompletionSystemMessageParam,
+) (bedrockSystem []*awsbedrock.SystemContentBlock, err error) {
```

Comment:
same, wasted slice.

Reply:
maybe my previous comment was bad - accepting pointer to the slice here would be better

Reply:
revert it back